src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewEngines/CompositeViewEngine.cs

```diff
@@ -77,6 +85,14 @@ public ViewEngineResult GetView(string executingFilePath, string viewPath, bool
                 throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(viewPath));
             }
 
+            if (ViewEngines.Count == 0)
+            {
+                throw new InvalidOperationException(Resources.FormatViewEnginesAreRequired(
+                    typeof(MvcViewOptions).FullName,
+                    nameof(MvcViewOptions.ViewEngines),
+                    typeof(IViewEngine).FullName));
+            }
```

Comment:
In contrast to the file provider part - I think this is right. If you get here it's because you tried to render a view, which will never succeed for your app.